### PR TITLE
Prepare carousel for A/B testing

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds the Firebase infrastructure in preparation of A/B testing and loads the Firebase Remote Config parameters on startup. The value is then stored in a DataStore but it's not being used yet as the survey A/B test will be added next sprint. However, I decided to include it because we might need A/B testing functionality for other experiments.

The other change here include:

- The carousel screens have been reordered as per updated designs
- New [carousel action tracking](https://wp.me/pe5sF9-dj#comment-376)
- Matched the onboarding display behavior [with iOS](https://github.com/woocommerce/woocommerce-ios/pull/7307)

**To test:**

- Verify that the screens are now ordered
image

- Verify that the analytics are being tracked as described by this table:
image

- **Display logic test**
_Prerequisite: if you have tested this branch before, please delete and reinstall the app to clear all user defaults._

1. Launch the app
2. Log out if needed --> login onboarding screen should be shown
3. Tap "Next" only twice to go to the last feature page on the onboarding screen so that it is not dismissed
4. Relaunch the app --> login onboarding screen should be shown again
5. Tap "Next" three times or tap "Skip" to go to the prologue screen
6. Relaunch the app --> login onboarding screen should not be shown
7. Log in to the app
8. Log out --> login onboarding screen should not be shown

### CODE REVIEW

File: WooCommerce/build.gradle [hunk 1]
```diff
@@ -173,6 +173,8 @@ dependencies {
 
     implementation 'com.google.firebase:firebase-messaging-ktx:23.0.5'
     implementation 'com.google.android.gms:play-services-auth:20.2.0'
+    implementation 'com.google.firebase:firebase-config-ktx:21.1.0'
+    implementation 'com.google.firebase:firebase-analytics-ktx:21.0.0'
```
Comment (wzieba) [inline WooCommerce/build.gradle]: ❓ Could we use [Firebase BoM](https://firebase.google.com/docs/android/learn-more#bom) instead?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/config/RemoteConfigManager.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.woocommerce.android.di.AppCoroutineScope
+import com.woocommerce.android.util.PackageUtils
+import com.woocommerce.android.util.WooLog
+import com.woocommerce.android.util.WooLog.T
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/config/RemoteConfigManager.kt]: 🔧 I think it doesn't have to be a singleton - it doesn't hold any state. Maybe we could remove it to save DI additional hustle?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/FirebaseTracker.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.woocommerce.android.analytics
+
+import com.google.firebase.analytics.ktx.ParametersBuilder
+import com.google.firebase.analytics.ktx.analytics
+import com.google.firebase.analytics.ktx.logEvent
+import com.google.firebase.ktx.Firebase
+import javax.inject.Inject
+
+class FirebaseTracker @Inject constructor() {
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/FirebaseTracker.kt]: WDYT about hiding the implementation behind an interface? I think `LoginActivity` should not know anything about "firebase".

**Review by wzieba — COMMENTED** [review]
👋 @0nko I hope you don't mind chiming in. Leaving a few comments, WDYT?

File: WooCommerce/build.gradle (hunk 1 above)
Comment (0nko) [reply WooCommerce/build.gradle]: Good idea 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/FirebaseTracker.kt (hunk 3 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/FirebaseTracker.kt]: Good call 👍

**Comment by 0nko** [comment]
Thanks for the review @wzieba, ready for another round.

**Review by wzieba — APPROVED** [review]
Thanks for the improvements, looks good to me!

**Pull request merged**